src/server/verification/verificationAPI.js

```diff
@@ -473,10 +482,12 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
       // initiate smart contract to send bonus to user
       try {
         txHash = await AdminWallet.redeemBonuses(currentUser.gdAddress, toRedeemInWei, {
-          onReceipt: r => {
+          onReceipt: async r => {
```

Comment:
there's no need for that.
receipt = await AdminWAllet.redeemBonuses()
read my previous request to change this again.

Reply:
@sirpy I understand what changes you require, but waiting for receipt is not a good idea because it may take a lot of time and the request will fail with timeout on the client app.

Reply:
you are right.
you still need to fix your code then
1. you have await before AdminWallet.redeem. it doesnt return a txhash it returns a receipt. so you need to remove this await
2. you need to onTxhash+onerror callbacks. ontxhash would finish the request ie res.json. onerror would call fail().

@StanislavShevchenko when we receive an Error from the blockchain do we still need to increment the nonce? i think we do, but I think fail() doesnt increment the nonce

Reply:
done

Reply:
@sirpy regarding nonce. We use lock for users address just to prevent him to get the same bonuses more than 1 time. So the nonce is not used in this scenario.